### PR TITLE
Deterministic DruidCoordinatorTest

### LINKED ISSUE #2117: Non-deterministic io.druid.server.coordinator.DruidCoordinatorTest testCoordinatorRun
Running io.druid.server.coordinator.DruidCoordinatorTest
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.251 sec <<< FAILURE! - in io.druid.server.coordinator.DruidCoordinatorTest
testCoordinatorRun(io.druid.server.coordinator.DruidCoordinatorTest)  Time elapsed: 3.118 sec  <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:555)
	at org.junit.Assert.assertEquals(Assert.java:542)
	at io.druid.server.coordinator.DruidCoordinatorTest.testCoordinatorRun(DruidCoordinatorTest.java:351)

### PR DESCRIPTION
**Pull request by pjain1**
Fixes #2117

### CODE REVIEW

File: server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java [hunk 1]
```diff
@@ -371,7 +372,9 @@ public void childEvent(
     Assert.assertNotNull(dataSourceMap.get(dataSource));
     // Simulated the adding of segment to druidServer during SegmentChangeRequestLoad event
     // The load rules asks for 2 replicas, therefore 1 replica should still be pending
-    Assert.assertEquals(1L, dataSourceMap.get(dataSource).get());
+    while(dataSourceMap.get(dataSource).get() != 1L) {
+      Thread.sleep(50);
```
Comment (drcrallen) [inline server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java]: needs timeout

**Comment by drcrallen** [comment]
Can this be finagled to use a latch instead?

File: server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java (hunk 1 above)
Comment (pjain1) [inline server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java]: there is a global timeout on the test

**Comment by pjain1** [comment]
@drcrallen not really as it depends on when coordinator wakes and update the replication status

**Comment by gianm** [comment]
👍

**Pull request merged**